lib/widgets/message_list.dart

```diff
@@ -155,6 +164,21 @@ class MessageListPage extends StatefulWidget {
         initNarrow: narrow, initAnchorMessageId: initAnchorMessageId));
   }
 
+  /// The "revealed" state of a message from a muted sender.
+  ///
+  /// This is updated via [MessageListPageState.revealMutedMessage]
+  /// and [MessageListPageState.unrevealMutedMessage].
+  ///
+  /// Uses the efficient [BuildContext.dependOnInheritedWidgetOfExactType],
+  /// so this is safe to call in a build method.
+  static RevealedMutedMessagesState revealedMutedMessagesOf(BuildContext context) {
+    final state =
+      context.dependOnInheritedWidgetOfExactType<_RevealedMutedMessagesProvider>()
+      ?.state;
+    assert(state != null, 'No _RevealedMutedMessagesProvider ancestor');
```

Comment:
nit: From the caller's perspective, I think the error is that there's no MessageListPage:
```suggestion
  static RevealedMutedMessagesState revealedMutedMessagesOf(BuildContext context) {
    final state =
      context.dependOnInheritedWidgetOfExactType<_RevealedMutedMessagesProvider>()
      ?.state;
    assert(state != null, 'No MessageListPage ancestor');
```

If there's somehow a MessageListPage but no _RevealedMutedMessagesProvider, that's a bug in MessageListPage.

This is similar to what PerAccountStoreWidget.of says:
```dart
    final widget = context.dependOnInheritedWidgetOfExactType<_PerAccountStoreInheritedWidget>();
    assert(_debugCheckFound(context, widget));
// …
  static bool _debugCheckFound(BuildContext context, Object? ancestor) {
    // Compare [debugCheckHasMediaQuery], and its caller [MediaQuery.of].
    assert(() {
      if (ancestor != null) return true;
      throw FlutterError.fromParts([
        ErrorSummary('No PerAccountStoreWidget ancestor found.'),
```

where the thing it actually looks up is _PerAccountStoreInheritedWidget but that's an implementation detail.